napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_buffer_info*
+
+#### Signature
+```
+napi_status napi_get_buffer_info(napi_env env,
+                                 napi_value value,
+                                 void** data,
+                                 size_t* length)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: napi_value representing the node::Buffer being queried
+- `[out] data`: The underlying data buffer of the node::Buffer
+- `[out] length`: Length in bytes of the underlying data buffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to retrieve the underlying data buffer of an node::Buffer and it's length.
+
+### *napi_get_global*
+
+#### Signature
+```
+napi_status napi_get_global(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: `napi_value` representing JavaScript Global Object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_null*
+
+#### Signature
+```
+napi_status napi_get_null(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: `napi_value` representing JavaScript Null Object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_prototype*
+
+#### Signature
+```
+napi_status napi_get_prototype(napi_env env, napi_value object, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  object`: `napi_value` representing JavaScript Object whose prototype to return
+- `[out] result`: `napi_value` representing prototype of the given object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_typedarray_info*
+
+#### Signature
+```
+napi_status napi_get_typedarray_info(napi_env env,
+                                     napi_value typedarray,
+                                     napi_typedarray_type* type,
+                                     size_t* length,
+                                     void** data,
+                                     napi_value* arraybuffer,
+                                     size_t* byte_offset)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  typedarray`: `napi_value` representing the TypedArray whose properties to query
+- `[out] type`: Scalar datatype of the elements within the TypedArray
+- `[out] length`: Number of elements in the TypedArray
+- `[out] data`: The data buffer underlying the typed array
+- `[out] byte_offset`: The byte offset within the data buffer from which to start projecting the TypedArray
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns various properties of a typed array. 
+Warning: This API is dangerous since the underlying data buffer is managed by the VM
```

Comment:
I wouldn't say the API itself is "dangerous", just that using the returned data buffer requires caution.